core/dbt/contracts/results.py

```diff
@@ -137,6 +147,15 @@ def write(self, path, omit_none=True):
         output = FreshnessRunOutput(meta=meta, sources=sources)
         output.write(path, omit_none=omit_none)
 
+    def __len__(self):
+        return len(self.results)
+
+    def __iter__(self):
+        return iter(self.results)
+
+    def __getitem__(self, idx):
+        return self.results[idx]
```

Comment:
These changes make results a lot like the old thing where we returned `result.results`, which meant I didn't have to update a million tests. Also, it just seemed like a nice thing.